database.go

```diff
@@ -0,0 +1,1235 @@
+}
+
+// Backup Resource Path
+func (bk *databaseBackup) resourcePath(backupID string) string {
+	return databaseBackupsResourcePath + "/" + backupID
+}
+
+// Backup Resource Create Path
+func (bk *databaseBackup) resourceCreatePath(resourceType string, resourceID string) string {
+	return "/" + resourceType + "/" + resourceID + databaseBackupsResourcePath
+}
+
+// Schedule Resource List Path
+func (sc *databaseSchedule) resourcePath(scheduleID string) string {
+	return databaseSchedulesResourcePath + "/" + scheduleID
+}
+
+// Schedule List In Resource Path
+func (sc *databaseSchedule) resourceOtherPath(resource string, resourceID string) string {
+	return "/" + resource + "/" + resourceID + databaseSchedulesResourcePath
+}
+
+// Schedule Resource Backup Path
+func (sc *databaseSchedule) resourceBackupListPath(scheduleID string) string {
+	return databaseSchedulesResourcePath + "/" + scheduleID + databaseBackupsResourcePath
+}
+
+// Au Resource Path
+func (au *databaseAutoscaling) resourcePath(resourceID string) string {
+	return databaseInstancesResourcePath + "/" + resourceID + databaseAutoscalingResourcePath
+}
+
+// fw Resource Path
+func (fw *databaseFirewall) resourcePath(nodeID string) string {
+	return databaseNodesResourcePath + "/" + nodeID + databaseTrustedSourcesResourcePath
+}
+
+// ta Resource Path
+func (ta *databaseTask) resourcePath(taskID string) string {
+	return databaseTasksResourcePath + "/" + taskID + "status"
+}
+
+// DatabaseListOption contains option when list resource
+type DatabaseListOption struct {
+	Page           int `json:"page,omitempty"`
+	ResultsPerPage int `json:"results_per_page,omitempty"`
+	Name           string `json:"name,omitempty"`
+}
+
+// DatabaseAutoscalingAction wrap information about autoscaling of database instance.
+type DatabaseAutoscalingAction struct {
+	Autoscaling DatabaseAutoscaling `json:"autoscaling"`
+}
+
+//DatabaseAutoscalingVolume contains volume information of autoscaling.
+type DatabaseAutoscalingVolume struct {
+	Threshold int `json:"threshold"`
+	Limited   int `json:"limited"`
+}
+
+//DatabaseAutoscaling contains autoscaling information of a database instance.
+type DatabaseAutoscaling struct {
+	Enable    bool                      `json:"enable"`
+	Volume    DatabaseAutoscalingVolume `json:"volume"`
+	Receivers []map[string]string       `json:"receivers,omitempty"`
+	Alarms    []map[string]string       `json:"alarms,omitempty"`
+}
+
+//DatabaseDNS contains DNS information.
+type DatabaseDNS struct {
+	Private string `json:"private"`
+	Public  string `json:"public"`
+}
+
+//DatabaseAddressesPrivate contains private addresses information of database.
+type DatabaseAddressesPrivate struct {
+	IPAddress   string `json:"ip_address"`
+	NetworkName string `json:"network_name"`
+}
+
+//DatabaseAddressesPublic contains public addresses information of database.
+type DatabaseAddressesPublic struct {
+	IPAddress   string `json:"ip_address"`
+	NetworkName string `json:"network_name"`
+}
```

Comment:
what difference between DatabaseAddressesPrivate and DatabaseAddressesPublic struct?